src/coreclr/nativeaot/System.Private.CoreLib/src/System/GC.NativeAot.cs

```diff
@@ -851,18 +844,12 @@ static T[] AllocateNewUninitializedArray(int length, bool pinned)
         /// <typeparam name="T">Specifies the type of the array element.</typeparam>
         /// <param name="length">Specifies the length of the array.</param>
         /// <param name="pinned">Specifies whether the allocated array must be pinned.</param>
-        /// <remarks>
-        /// If pinned is set to true, <typeparamref name="T"/> must not be a reference type or a type that contains object references.
-        /// </remarks>
         public static unsafe T[] AllocateArray<T>(int length, bool pinned = false)
         {
             GC_ALLOC_FLAGS flags = GC_ALLOC_FLAGS.GC_ALLOC_NO_FLAGS;
 
             if (pinned)
             {
-                if (RuntimeHelpers.IsReferenceOrContainsReferences<T>())
-                    ThrowHelper.ThrowInvalidTypeWithPointersNotSupported(typeof(T));
-
                 flags = GC_ALLOC_FLAGS.GC_ALLOC_PINNED_OBJECT_HEAP;
             }
 
```

Comment:
Nit: You can replace `EETypePtr.EETypePtrOf<T[]>().RawValue` below with just `MethodTable.Of<T[]>()` while you are on it. It is more terse way to get the MethodTable of specific type in nativeaot.